File: jan-14-25/sub.java
class Solution {
    public int[] findThePrefixCommonArray(int[] A, int[] B) {
        int n = A.length;
        int[] mp = new int[n+1];
        int[] ans = new int[n];

        int com = 0;

        for(int i = 0; i<n; i++){
            mp[A[i]]++;
            if(mp[A[i]] == 2) com++;
            mp[B[i]]++;
            if(mp[B[i]] == 2) com++;
            ans[i] = com;
        }

        return ans;
    }
}

File: jan-14-25/sub.py
from typing import List

class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        n = len(A)
        mp = [0] * (n + 1)

        comm = 0
        ans = [0] * n

        for i in range(n):
            mp[A[i]] += 1
            if mp[A[i]] == 2:
                comm += 1
            mp[B[i]] += 1
            if mp[B[i]] == 2:
                comm += 1

            ans[i] = comm

        return ans